# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1484600418 18000
#      Mon Jan 16 16:00:18 2017 -0500
# Node ID b443bfa3bfea2a1d6eb9829aa226b8f945973420
# Parent  f97e22a9012a44d01e3b95fb6cd9f6da10aa92b1
# Parent  7f91376af4d3f3e330b5b6457e29d7cb1e131c69
maint: Merge stable to default.

diff --git a/libinterp/octave-value/ov-oncleanup.cc b/libinterp/octave-value/ov-oncleanup.cc
--- a/libinterp/octave-value/ov-oncleanup.cc
+++ b/libinterp/octave-value/ov-oncleanup.cc
@@ -93,17 +93,18 @@ octave_oncleanup::~octave_oncleanup (voi
     {
       std::string msg = last_error_message ();
       warning ("onCleanup: error caught while executing cleanup function:\n%s\n",
                msg.c_str ());
 
     }
   catch (const octave::exit_exception&)
     {
-      throw;
+      // This shouldn't happen since we disabled quit above.
+      warning ("onCleanup: exit disabled while executing cleanup function");
     }
   catch (...) // Yes, the black hole.  We're in a d-tor.
     {
       // This shouldn't happen, in theory.
       warning ("onCleanup: internal error: unhandled exception in cleanup action");
     }
 }
 
